### PR TITLE
[serve] Unify handle option setting

### PR DESCRIPTION
**Pull request by edoakes**
## Why are these changes needed?

We had a few separate codepaths for updating `HandleOptions`, each of which was hard to follow. Unified it into a method on the `HandleOptions` class itself.

I also fly-by fixed a bug related to multiplexing: we were setting the model ID in the request context rather than in handle options, which meant it would also affect subsequent `serve.get_multiplexed_model_id()` calls on the caller. See the test case in `test_multiplex.py`.

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by edoakes** [comment]
@GeneDer follow-up from the previous PR

**Review by GeneDer — APPROVED** [review]
:shipit:
